### ISSUE TITLE
Update upstream repository?

### ISSUE THREAD

**Issue by robert7** [artmg/nixnote2-packaging#31]
Hi Art, 
would you be willing to update upstream repository to my fork? (You wrote in https://github.com/baumgarr/nixnote2/issues/446 that this would be possible). Although I did an AppImage build, it seems the Ubuntu PPA would still be helpful to many people. Also the themeing currently doesn't work with AppImage, but will work with native package.

Currently the version is "v2.1.0-beta2".. it should be already quite stable - at least for the functions I'm using. So you could update the daily-build to use "master" branch from my repository. I wrote a [migration info for v2.0](https://github.com/robert7/nixnote2/wiki/Migration-v2.0-to-v2.1) users, but basically nothing needs to be done when switching from v2.0 to v2.1 - all should work out of the box and the new version should pickup the existing configs and data. Also switch from v2.1 back to v2.0 is possible without problems.

I did some changes in the build, so your script will need updating. But I will gladly assist you. Mainly [html tidy](https://github.com/htacg/tidy-html5) must be used with 5.6.0 tag and Qt4 build is not supported anymore. It is best to use Qt 5.5.1 (Qt4 or Qt5.6+ will currently not work).

How to build is described in [README.md](https://github.com/robert7/nixnote2/blob/master/README.md), but most up to date is the docker file "[development/docker/Dockerfile.ubuntu_xenial](https://github.com/robert7/nixnote2/blob/master/development/docker/Dockerfile.ubuntu_xenial)" which I'm using to build the AppImage. There you see all required dependencies and how to add tidy.

**Comment by kapad** [comment]
As a user, I totally agree with @robert7 that updating the upstream repo for the ubuntu ppa would be the best solution for me to switch over to using @robert7's repo, which is now the only one which is being maintained.

**Comment by robert7** [comment]
Update: I found out, that I misunderstanded how the packaging works; so simple switch now is not possible. I first need to fix the packaging scripts. But anyway, let we now whenever you want to update it.

**Comment by artmg** [comment]
@robert7, I am not sure that I would feel happy simply switching users from one codebase to the next without them actively subscribing to such a change. Aside from trust issues, there is also the matter of testing builds from your codebase before any cutover.

The _PPA_ (Personal Packaging Archive) that users of @baumgarr code have already added is called **nixnote/nixnote2-daily** and I would propose to **create a new _recipe_ with a new name**. If users decide to move with your fork of nixnote2, then they would have to _actively subscribe_ to your updates by simply removing the old PPA and adding the new one. 

This would also be a sensible point to consider the following matters for your forked project:
* **branding**: whether you might vary the name of your fork (e.g. nixnote3 or something more significantly different)
* **release management**: are there any very active users who might be willing to support testing by subscribing to a daily build, allowing you to create separate stable releases. 
* **membership**: will you remain as sole developer with rights to update code and accept pull requests or do you have other members willing to join a team

I have designed the Launchpad account to be generic, and once any changes are set up I am happy to consider transferring ownership to any team you might establish.

In the meantime, if anyone wants to understand how the existing Launchpad PPA has been put together, I put the instructions onto [this repo's wiki](https://github.com/artmg/nixnote2-packaging/wiki)

Please let me know how you'd like to move forward
Cheers
Art

**Comment by robert7** [comment]
Hi Art,
after reconsideration I agree with you, that switching the upstream for the existing recipe would be a bad idea. So let not do this.

I did read your wiki pages yesterday and couple of related packages and I still don't fully understand, how it works... and currently because of limited time I have for the project, its out of scope for me. Therefore if you would be willing to create new "recipe" with new name, which would build from my repository, this would be **great help**.

The idea about switching the repo for the existing recipe was following: I think most people find "nixnote" someway, but then see the project is "dead" and simply go away, they don't even find out that there is new updated version. Which I find a bit pity. But never mind.

* **name**: I did not changed that much in the project, so I think name "nixnote21" could be a good option for now (for version "v.2.1").  
* **release management**: I think for now it would be enough if you could setup a "nixnote/nixnote21-daily" which would build from master branch (later maybe "nixnote/nixnote2-stable" but now it would be too early)
* **membership**: although I appreciate any help, currently there is no team. it's just me. but anyone is invited to join :) of course pull requests or any other kind of help is appreciated

Regarding building:
* what I wrote in the first comment regarding the Qt 5.5 version is not correct. It should build with the system qt5 from Ubuntu 16.04 to 18.04 without problems (same as in upstream). So this should be no problem. (Build in 14.04 would be possible, but I'm not sure if it is worth it. I will try to create the AppImage in 14.04, so PPA would be enough starting from 16.04).
* one difficulty comparing to upstream is, that now "libtidy" v5.6 is needed, this isn't currently available as package. I'm not sure how this is to be implemented. Your recipe now contains "libtidy-dev" which is 0.99 in 16.04 and 5.2 in 18.04.  We can discuss the possible options via email or here (in case you will be willing to create the recipe).

cheers
Robert

**Comment by robert7** [comment]
Hi Art,
so I created my own PPA at https://launchpad.net/~robert7/+archive/ubuntu/nixnote21
I forked your packaging repository and made required changes.

I never created a debian pkg before - so I was completely lost at beginning..
Although I needed to tinker it a lot... your existing packaging was a great help. Great thanks!

**Comment by artmg** [comment]
Hi Robert,

I'm very pleased to hear that you managed to create your PPA. Launchpad does seem a little odd at first which is why I made sure that I took copious notes in the [wiki](https://github.com/artmg/nixnote2-packaging/wiki). 

It makes great sense for the same person to control both the project and its PPA, and maintenance of the packaging source is as simple as updating dependency changes and including new distribution series. I left mine as a daily build, so it would automatically repackage based on source-code commits, but its up to you how to do your own release management.

Good luck with the project
All the best
Art
